src/cloud_provider/kubernetes.rs

```diff
@@ -704,6 +732,38 @@ pub fn check_kubernetes_has_enough_resources_to_deploy_environment(
     Ok(())
 }
 
+/// check that there is enough nodes before any charts deployment.
+pub fn check_kubernetes_has_enough_nodes_to_deploy<P>(
+    kubernetes_config: P,
+    envs: Vec<(&str, &str)>,
+    max_nodes: usize,
+) -> Result<(), SimpleError>
+where
+    P: AsRef<Path>,
+{
+    return match kubectl_exec_get_node(kubernetes_config, envs) {
+        Err(e) => {
+            error!("Can't get nodes before services deployments");
+            Err(e)
+        }
+        Ok(nodes) => match nodes.items.len() < (max_nodes * 80 / 100) {
```

Comment:
doesn't it break autoscaler ?
if you have a cluster with [3 min, 10 max] nodes,
and say you are at 3 nodes ATM, you will never be able to scale your cluster no ?